packages/smooth_app/lib/pages/product/new_product_page.dart

```diff
@@ -0,0 +1,271 @@
+import 'package:smooth_app/pages/product/common/product_dialog_helper.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+
+class NewProductPage extends StatefulWidget {
+  const NewProductPage({
+    required this.product,
+  });
+
+  final Product product;
+
+  @override
+  State<NewProductPage> createState() => _ProductPageState();
+}
+
+class _ProductPageState extends State<NewProductPage> {
+  late Product _product;
+  bool _first = true;
+
+  @override
+  void initState() {
+    super.initState();
+    _updateHistory(context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    if (_first) {
+      _first = false;
+      _product = widget.product;
+    }
+    return Scaffold(
+      backgroundColor: SmoothTheme.COLOR_PRODUCT_PAGE_BACKGROUND,
+      appBar: AppBar(
+        title: Text(_getProductName(appLocalizations)),
+        actions: <Widget>[
+          PopupMenuButton<String>(
+            itemBuilder: (final BuildContext context) =>
+                <PopupMenuEntry<String>>[
+              PopupMenuItem<String>(
+                value: 'web',
+                child: Text(appLocalizations.label_web),
+              ),
+              PopupMenuItem<String>(
+                value: 'refresh',
+                child: Text(appLocalizations.label_refresh),
+              ),
+            ],
+            onSelected: (final String value) async {
+              switch (value) {
+                case 'web':
+                  LaunchUrlHelper.launchURL(
+                      'https://openfoodfacts.org/product/${_product.barcode}/',
+                      false);
+                  break;
+                case 'refresh':
+                  final ProductDialogHelper productDialogHelper =
+                      ProductDialogHelper(
+                    barcode: _product.barcode!,
+                    context: context,
+                    localDatabase: localDatabase,
+                    refresh: true,
+                  );
+                  final Product? product =
+                      await productDialogHelper.openUniqueProductSearch();
+                  if (product == null) {
+                    productDialogHelper.openProductNotFoundDialog();
+                    return;
+                  }
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBar(
+                      content: Text(appLocalizations.product_refreshed),
+                      duration: const Duration(seconds: 2),
+                    ),
+                  );
+                  _product = product;
+                  await _updateHistory(context);
+                  break;
+                default:
+                  throw Exception('Unknown value: $value');
```

Comment:
Could we restructure the code to avoid having to handle this exception? One way would be to use an enum instead of strings.

Reply:
Well we definitely need the exception, even with enums, it should throw when someone adds a new enum value without adding a handler for it. 

That said, using enums is a much cleaner way to code this than strings, updated code to use enums instead.